Source/Core/Core/Core.cpp

```diff
@@ -993,9 +1008,38 @@ void Shutdown()
   HostDispatchJobs();
 }
 
-void SetOnStateChangedCallback(StateChangedCallbackFunc callback)
+int AddOnStateChangedCallback(StateChangedCallbackFunc callback)
 {
-  s_on_state_changed_callback = std::move(callback);
+  for (int i = 0; i < s_on_state_changed_callbacks.size(); ++i)
```

Comment:
```suggestion
  for (size_t i = 0; i < s_on_state_changed_callbacks.size(); ++i)
```

Otherwise this will cause a sign-compare warning

Reply:
Good catch, thanks.